business/istio_status.go

```diff
@@ -189,11 +228,11 @@ func (iss *IstioStatusService) getStatusOf(ds []apps_v1.Deployment) (IstioCompon
 		// Component found
 		cf[appLabel] = true
 
-		if status := GetDeploymentStatus(d); status != Healthy {
+		if wl.Status != Healthy {
```

Comment:
This logic is linked with my previous comment.
Basically it's duplicating what business workloads is doing, so I propose to use that instead and unify.

Perhaps we can discuss if we want to do it in this or other PR, but I'm sure we need to move in that direction.